xmrstak/backend/cpu/minethd.cpp

```diff
@@ -317,23 +308,17 @@ minethd::cn_hash_fun minethd::func_selector(bool bHaveAes, bool bNoPrefetch)
 
 void minethd::pin_thd_affinity()
 {
-	//Lock is needed because we need to use oWorkThd
-	std::lock_guard<std::mutex> lock(work_thd_mtx);
-
 	// pin memory to NUMA node
 	bindMemoryToNUMANode(affinity);
-
-#if defined(__APPLE__)
-	printer::inst()->print_msg(L1, "WARNING on MacOS thread affinity is only advisory.");
-#endif
-	thd_setaffinity(oWorkThd.native_handle(), affinity);
 }
 
 void minethd::work_main()
 {
 	if(affinity >= 0) //-1 means no affinity
 		pin_thd_affinity();
```

Comment:
`bindMemoryToNUMANode(affinity);` should be called here directly and `pin_thd_affinity` can be removed.

Reply:
Ok. I will test that one on Windows too.